src/shared/s3_manager.py

```diff
@@ -64,6 +75,18 @@ def __init__(self, event):
                 f"{self.study}__{self.data_package}__{self.site}__{self.version}/"
                 f"{self.study}__{self.data_package}__{self.site}__flat.parquet"
             )
+        if study:
+            self.study = study
+        if site:
+            self.site = site
+        if data_package:
+            self.data_package = data_package
+        if version:
+            self.version = version
+        if site and study:
+            self.lockfile = (
+                f"{enums.BucketPath.META.value}/lockfiles/{self.site}__{self.study}.json"
+            )
```

Comment:
This pattern here (and above) is setting a trap for yourself, in my mind - I think you want to unconditionally set `self.lockfile` to *something*, even if just `None`. Right now if you aren't setting it, it can be an attribute error to reference `self.lockfile`, and all code that does has to be careful.

Maybe something as dumb as:
```python
        self.lockfile = (
            f"{enums.BucketPath.META.value}/lockfiles/{self.site}__{self.study}.json"
        ) if site and study else None
```

Or just do `self.lockfile = None` before the if block.